app/components/Account/AccountVoting.jsx

```diff
@@ -478,9 +478,13 @@ class AccountVoting extends React.Component {
     _getWorkerArray() {
         let workerArray = [];
 
-        for (let i = 0; i < 100; i++) {
+        let maxId = 100;
+        for (let i = 0; i < maxId; i++) {
             let id = "1.14." + i;
             let worker = ChainStore.getObject(id, false, false);
+            if (i <= maxId) {
+                maxId = i+10;
```

Comment:
This code is really ugly. We can use `while (true)` directly. Or `for (let i = 0; true; i++)`.

By the way, we do need to improve the core API to better serve UI.

Reply:
Thanks for the note. Your solution is more elegant yes.

There is also the call get_all_workers that can be used, but I hadn't the time at the point.

Reply:
Why have to use break and not pull the first worker assignment out of the loop and simply test for worker!==null ?

Reply:
Not sure why a break on null isn't sufficient. This still should be changed to use get_all_workers instead.

Reply:
It is sufficient,  just avoiding breaks tends to be clearer imho .